### PR TITLE
[WIP] Add class_schedule page

### PR DESCRIPTION
**Pull request by palak199**
I am trying to use bootstrap's collapse to toggle the ```add new schedule``` however it is not working.
```<div class="col-md-2 mb-3">
                <button class="btn btn-primary" type="button" data-bs-toggle="collapse" data-bs-target="#collapseExample" aria-expanded="false" aria-controls="collapseExample">
                    add new
                  </button>
            </div>

 <div class="collapse" style="display: none" id="collapseExample">
        <div class="card card-body">
            the card that should open, but doesn't :(
        </div>
    </div> 
```
these lines contain the code for collapse

### CODE REVIEW

File: templates/home/class_schedule.html [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                    Add new
+                  </button>
+            </div>
+        </div>
+        <div class="collapse bg-light mt-3" id="collapseExample">
+            <div class="card card-body">
+                <h5 class="card-title text-center">Add new schedule</h5>
+                <div class="row mt-2 justify-content-center align-items-center">
+                    <div class="col-md-3 mb-3">
+                        <label for="sch-day">Day wise</label>
```
Comment (garg3133) [inline templates/home/class_schedule.html:53]: just keep this `Day`, `Section` and `Subject`.

**Comment by palak199** [comment]
Thank you for reviewing, I have made the changes.

File: home/views.py [hunk 2]
```diff
@@ -190,4 +190,36 @@ def ajax_dashboard(request):
...
+    subjects = Schedule.SUBJECT
+    data = {}
+    #sections = Schedule.objects.order_by('section__section_id')
+    schedule = Schedule.objects.order_by('day', 'section__section_id', 'subject' )
+    sections = Section.objects.exclude(schedule=None).order_by('section_id')
+
+    #print(sections)
+    #for section in sections:
+        #data[section.section.section_id] = section.section.name 
+
```
Comment (garg3133) [inline home/views.py:216]: Can you please remove all these comments?

File: home/views.py [hunk 3]
```diff
@@ -190,4 +190,36 @@ def ajax_dashboard(request):
...
+    #for section in sections:
+        #data[section.section.section_id] = section.section.name 
+    
+    context={
+        'days': days,
+        'subjects': subjects,
+        #'sections': data,
+        'sections': sections,
+        'schedule': schedule
+}
```
Comment (garg3133) [inline home/views.py:223]: Indentation of this curly brace is still not proper.

File: templates/home/class_schedule.html [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+                    <option value="{{ day }}">{{ day_name }}</option>
+                    {% endfor %}
+                </select>
+            </div>
+            <div class="col-md-3 mb-3">
+                <label for="sch-section">Section</label>
+                <select name="section" class="custom-select d-block " id="sch-section">
+                    <option value="">Select Section</option>
+                    {% for section in sections %}
+                    <option value="{{ section }}">{{ section }}</option>
```
Comment (garg3133) [inline templates/home/class_schedule.html:29]: You should use `section.section_id` in value attribute and `section.name` in the HTML part.

File: templates/home/class_schedule.html [hunk 5]
```diff
@@ -50,28 +50,28 @@ <h3 class="text-center">Class Schedule</h3>
...
                     </div>
                     <div class="col-md-3 mb-3">
-                        <label for="sch-section">Section wise</label>
+                        <label for="sch-section">Section</label>
                         <select name="section" class="custom-select d-block w-100" id="sch-section">
                             <option value="">Select Section</option>
-                            {% for section, section_name in section %}
-                            <option value="{{ section }}">{{ section_name }}</option>
+                            {% for section in sections %}
+                            <option value="{{ section }}">{{ section }}</option>
```
Comment (garg3133) [inline templates/home/class_schedule.html:66]: Do the same thing here also, as mentioned above.

Also, here all sections should be visible and not just the sections which are being used in some schedule. For this, you can do something like this:

```
all_sections = Section.objects.order_by('section_id')
active_sections = all_sections.exclude(schedule=None)
```
and use `all_sections` at this place and `active_sections` above.

**Comment by garg3133** [comment]
@palak199 You seem to be busy somewhere, so I'm merging this PR after making the above-said changes myself.

Please make a pull when you start to work on this again :))

**Pull request merged**